### PR TITLE
feature(colors): initial colors organism

### PR DESCRIPTION
**Pull request by rhelling**
- [x] Add unit test (Robbert)
- [x] Make keyboard accessible
- [x] Refactor transparency to new ProductColor props
- [x] Reset c-image loaded state when image has changed
- [x] Verify overflow with a lot of colors

From Rework styling and classes
- [x] Clean up CSS, remove unnecessary extends
- [x] Add keyboard navigation
- [x] Add proper focus styling

I'm not sure if `Enter` should also change the active colour. Opinions welcome.

### CODE REVIEW

File: src/js/components/organisms/colors/stories/colors.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Colors
+
+## DSM
```
Comment (dan-entous) [inline src/js/components/organisms/colors/stories/colors.md]: could link to these instead with the current caveat:
below are links to the dsm guidelines, however, the current implementation does not include the tooltip show in the guidelines
* [small](https://ultimaker.invisionapp.com/dsm/ultimaker/ultimaker-com/asset/components/5c4af1b9dd18e10fa57787f7)
* [medium](https://ultimaker.invisionapp.com/dsm/ultimaker/ultimaker-com/asset/components/5c4b0f186b659fc43d6d11b3)
* [large](https://ultimaker.invisionapp.com/dsm/ultimaker/ultimaker-com/asset/components/5c4b13a4198d49e1eea97263)

File: src/js/components/organisms/colors/stories/colors.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+* [organisms/colors](https://ultimaker.invisionapp.com/dsm/ultimaker/ultimaker-com/asset/components/)
```
Comment (dan-entous) [inline src/js/components/organisms/colors/stories/colors.md]: could add design links:
## visual designs
* [small](https://ultimaker.invisionapp.com/d/main#/console/16453577/341178746/preview)
* [medium](https://ultimaker.invisionapp.com/d/main#/console/16453577/341178747/preview)
* [large](https://ultimaker.invisionapp.com/d/main#/console/16453577/341178748/preview)

## design specs
* [small](https://ultimaker.invisionapp.com/d/main#/console/16453577/344370029/preview)
* [medium](https://ultimaker.invisionapp.com/d/main#/console/16453577/344370030/preview)
* [large](https://ultimaker.invisionapp.com/d/main#/console/16453577/344370031/preview)

File: src/js/components/organisms/colors/colors.vue.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<article class="colors">
```
Comment (dan-entous) [inline src/js/components/organisms/colors/colors.vue.html]: i'm a bit confused about two things:

1. is it an organism or a molecule?
1. should it be its own organism or a molecule/organism inserted into an overview organism (src/js/components/organisms-page/overview/overview.html)?

seems to me we should be re-using the existing overview component, which is the basis for all of our organism components

File: src/js/components/organisms/colors/data/colors.data.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    default: {
+        title: 'Lorem ipsum .. enzo nog veel meer',
+        subtitle: 'moest er in van Martin',
+        colorList: [
+            {
+                rgbHex: '#ff1493',
+                exampleImage: {
+                    alt: 'Ultimaker Tough PLA print',
+                    mimeType: 'image/jpeg',
+                    type: 'Image',
```
Comment (dan-entous) [inline src/js/components/organisms/colors/data/colors.data.ts]: think the data should be using "real" data such as:

* https://ultimaker.com/img/materials/samples/PLA/Ultimaker-PLA-Blue@2x.png
* https://ultimaker.com/img/materials/samples/PLA/Ultimaker-PLA-Magenta@2x.png

Comment (rhelling) [reply src/js/components/organisms/colors/data/colors.data.ts]: Let discuss this globally, images on ultimaker.com do not resize with the c-image component. The svg-mocks do. Then the question arises where to host the images and how to work without a working internet connection on your local development environment.

File: src/js/components/molecules/color/color.ts [hunk 5]
```diff
@@ -24,13 +28,17 @@ export default class Color extends Vue implements ColorProps {
...
-        if (this.background === '') {
-            styles['color'] = 'black';
+        styles['color'] = ColorUtil.lightness(this.color) >= 0.5 ? '#000' : '#FFF';
 
-            return styles;
+        return styles;
+    }
+
+    get color() {
+        const hexColorNoPrefix = (/^([A-Fa-f0-9]{6})/);
```
Comment (zwaardje) [inline src/js/components/molecules/color/color.ts]: Kinda lame this needs to be verified in the frontend. Could the backend not check if the value has or needs a #? Like integration service?

File: src/js/components/organisms/colors/colors.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        display: flex;
+        flex-wrap: wrap;
+        list-style: none;
+        margin: 0 0 ($buildingUnit * 1.5);
+        padding: 0;
+
+        @include min-tablet {
+            margin-bottom: ($buildingUnit * 2) - 1px;
+        }
+
```
Comment (zwaardje) [inline src/js/components/organisms/colors/colors.scss]: 👾

File: src/js/components/organisms/colors/colors.scss [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        @include micro-animate(opacity);
+        @include max-mobile-xl {
+            margin-bottom: $buildingUnit * 3;
+        }
+
+        &--loading {
+            opacity: 0;
+        }
+    }
+
```
Comment (zwaardje) [inline src/js/components/organisms/colors/colors.scss]: 👾

File: src/scss/atomic/molecules/_container.scss [hunk 8]
```diff
@@ -1,6 +1,6 @@
 .container {
     max-width: $desktopHdWidth;
-    margin: 0 auto ($buildingUnit * 5);
+    margin: 0 auto;
```
Comment (zwaardje) [inline src/scss/atomic/molecules/_container.scss]: Doesn't this effect other containers?

File: src/scss/atomic/organisms/_hero.scss [hunk 9]
```diff
@@ -179,4 +179,4 @@ $imageLinkSizeDesktop: $buildingUnit * 10;
         @extend %reset-last-link-large-margin;
     }
 
-    */
\ No newline at end of file
+    */
```
Comment (zwaardje) [inline src/scss/atomic/organisms/_hero.scss]: Commenting what? Either remove or remove the code, github will remember.

File: src/js/components/atoms/c-image/c-image.ts [hunk 10]
```diff
@@ -56,15 +55,27 @@ export default class CImage extends Mixins(InView) implements ICImageProps {
...
     }
 
     beforeDestroy() {
         this.viewportUtil.removeResizeHandler(this.resizeHandler);
         this.viewportUtil.removeScrollHandler(this.resizeHandler);
     }
 
+    @Watch('url')
+    async urlChanged() {
+        // replaces previous image with tinyGif
```
Comment (zwaardje) [inline src/js/components/atoms/c-image/c-image.ts]: Code should explain what it does, comments should not.

File: src/js/components/molecules/color/color.ts (hunk 5 above)
Comment (rhelling) [reply src/js/components/molecules/color/color.ts]: Had a discussion about it, my opinion was exactly the same, maybe you can discuss this with @evdtempel? I suggest to leave this hack in the view model for now.

File: src/js/components/organisms/colors/colors.scss (hunk 6 above)
Comment (rhelling) [reply src/js/components/organisms/colors/colors.scss]: :rocket:

File: src/js/components/organisms/colors/colors.scss (hunk 7 above)
Comment (rhelling) [reply src/js/components/organisms/colors/colors.scss]: :moon:

File: src/scss/atomic/molecules/_container.scss (hunk 8 above)
Comment (rhelling) [reply src/scss/atomic/molecules/_container.scss]: I checked all organisms ;-)

File: src/js/components/atoms/c-image/c-image.ts (hunk 10 above)
Comment (rhelling) [reply src/js/components/atoms/c-image/c-image.ts]: I wanted to take a look at this outside of this feature request.

**Pull request merged**